prevent double snap

reco opened the pull request:
add new setting prevent-double-snap. is set to true prevent to do 2 snaps with a short snap-duration.

reco committed:
prevent double scroll

reco commented:
you can see an example here... https://moniquepean.imago.io

joelmukuthu commented:
This is really nice, thanks! Since this is a fix for a known bug I think we can adopt this as the default behaviour so no need for the new attribute, what do you think? 

`dist/snapscroll.js` is a generated file, so you'll have to move that code to `src/directives/snapscroll.js` so that it doesn't get overwritten the next time I'm releasing a new version. I'm also interested in seeing what effect this change will have on the tests. If [you'll allow me to](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) I can take care of that :)

joelmukuthu commented on dist/angular-snapscroll.js:
```diff
@@ -453,12 +473,20 @@
                     function bindWheel() {
                         wheelie.bind(element, {
                             up: function (e) {
+                                if (Math.abs(e.wheelDelta) > scope.oldWheelDelta ) {
```
Shouldn't this be `Math.abs(e.wheelDelta) < scope.oldWheelDelta`?

reco replied:
i thought that too before. but actually we should allow another snap when another scroll is done which increases the delta.

joelmukuthu commented:
I'm also wondering if this fix qualifies to be in https://github.com/joelmukuthu/angular-wheelie..

reco commented:
i would put it into the snapscroll. wheelie does in my opinion what it needs to do. leave this up to you.

ok i will remove the new attribute add it to the src. sure feel free to merge it test sure.. all yours

reco committed:
no extra attribute.

reco commented:
hey joelmukuthu i modified the src file. i cannot get the tests to pass though.

joelmukuthu commented:
i had a strong suspicion that that would happen. i'll look into that

Commit:
Refactor preventSnap into preventUp and preventDown

Makes it easier to manage when to toggle the other one instead of
relying of mouse wheel delta

joelmukuthu commented:
Hey @reco, sorry for the silence. I broke up `preventSnap` into `preventUp` and `preventDown` and simplified the toggling. PTAL and let me know if it still does the same thing as you intended. Btw that change got the failing tests to go down to from 16 to 9 :)

Commit:
Fix failing tests

Commit:
Allow configuring preventDoubleSnapDelay

Commit:
Update docs for preventDoubleSnapDelay

joelmukuthu commented:
@reco, this is now ready to be reviewed/merged. could you test the new built file in `dist/angular-snapscroll.min.js` against your site to see if it works as expected?

joelmukuthu commented:
Rolling with it :)
